client/src/app/services/dashboard.service.spec.ts

```diff
@@ -0,0 +1,55 @@
+
+import { TestBed } from '@angular/core/testing';
+import { DashboardService } from './dashboard.service';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+
+describe('CoursesService', () => {
+  let httpTestingController: HttpTestingController;
+  let service: DashboardService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [DashboardService],
+      imports: [HttpClientTestingModule]
+    });
+
+    httpTestingController = TestBed.inject(HttpTestingController);
+    service = TestBed.inject(DashboardService);
+  });
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('returned Observable should match the right data', () => {
+    const mockUser = {
+      user_name: 'a',
+      email: 'a@gmail.com',
+      first_name: 'a',
+      last_name: 'a',
+      role: 'SA'
+    };
+
+    service.getAllClients()
+      .subscribe(jsonData => {
+        const data = JSON.parse(jsonData);
+        expect(data.user_name).toEqual('a');
```

Comment:
Maybe access the mockUsers's members instead of typing the strings out again (like you did in the other test) ?
Like this:  mockUser.user_name